statsmodels/tsa/vector_ar/var_model.py

```diff
@@ -601,13 +620,23 @@ def fit(self, maxlags=None, method='ols', ic=None, trend='c',
 
         k_trend = util.get_trendorder(trend)
         self.exog_names = util.make_lag_names(self.endog_names, lags, k_trend)
-        self.nobs = len(self.endog) - lags
+        self.nobs = self.n_totobs - lags
+
+        # add exog to data.xnames (necessary because the length of xnames also
+        # determines the allowed size of VARResults.params)
+        if self.exog is not None:
+            x_names_to_add = [("exog%d" % i)
+                              for i in range(self.exog.shape[1])]
+            self.data.xnames = self.data.xnames[:k_trend] + \
+                               x_names_to_add + \
+                               self.data.xnames[k_trend:]
```

Comment:
We had problems/bugs with this pattern before, in ARMA(X) IIRC.
If `fit` is called repeatedly, then this will insert the `x_names_to_add` each time.
Can this be done in `__init__` instead?